pkg/kubelet/dockershim/convert.go

```diff
@@ -0,0 +1,91 @@
+/*
+Copyright 2015 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package dockershim
+
+import (
+	"fmt"
+	"strings"
+
+	dockertypes "github.com/docker/engine-api/types"
+	runtimeApi "k8s.io/kubernetes/pkg/kubelet/api/v1alpha1/runtime"
+)
+
+// This file contains helper functions to convert docker API types to runtime
+// API types, or vice versa.
+
+const (
+	// Status of a container returned by docker ListContainers
+	statusRunningPrefix = "Up"
+	statusCreatedPrefix = "Created"
+	statusExitedPrefix  = "Exited"
+)
+
+func toRuntimeAPIImage(image *dockertypes.Image) (*runtimeApi.Image, error) {
+	if image == nil {
+		return nil, fmt.Errorf("unable to convert a nil pointer to a runtime API image")
+	}
+
+	size := uint64(image.VirtualSize)
+	return &runtimeApi.Image{
+		Id:          &image.ID,
+		RepoTags:    image.RepoTags,
+		RepoDigests: image.RepoDigests,
+		Size_:       &size,
+	}, nil
+}
+
+func toRuntimeAPIContainer(c *dockertypes.Container) *runtimeApi.Container {
+	state := toRuntimeAPIContainerState(c.Status)
+	return &runtimeApi.Container{
+		Id:       &c.ID,
+		Name:     &c.Names[0],
+		Image:    &runtimeApi.ImageSpec{Image: &c.Image},
+		ImageRef: &c.ImageID,
+		State:    &state,
+		Labels:   c.Labels,
+	}
+}
+
+func toDockerContainerStatus(state runtimeApi.ContainerState) string {
+	switch state.String() {
```

Comment:
Uh..I can just use the runtimApi.ConainerState types in the `switch`. Don't know why I didn't do that. Fixed.